core/src/main/java/jenkins/model/lazy/AbstractLazyLoadRunMap.java

```diff
@@ -99,6 +106,145 @@
     private volatile Index index = new Index();
     private LazyLoadRunMapEntrySet<R> entrySet = new LazyLoadRunMapEntrySet<>(this);
 
+    private transient Set<Integer> keySet;
+    private transient Collection<R> values;
```

Comment:
```suggestion
    private transient volatile Set<Integer> keySet;
    private transient volatile Collection<R> values;
```

since you are not synchronizing on their initializations?

Reply:
Sure, why not. Though I note that [`AbstractMap` itself shares the same flaw](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/util/AbstractMap.java#L303-L328).